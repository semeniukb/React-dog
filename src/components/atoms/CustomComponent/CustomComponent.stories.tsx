import type { Meta, StoryObj } from '@storybook/react';
import { CustomComponent } from '.';

const meta = {
  title: 'Atoms/Custom Component',
  component: CustomComponent,
  argTypes: {
    variant: {
      options: ['contained', 'outlined'],
      control: 'radio'
    }
  }
} satisfies Meta<typeof CustomComponent>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    variant: 'contained',
    children: 'Button'
  }
};

export const Secondary: Story = {
  args: {
    variant: 'outlined',
    children: 'Button'
  }
};
