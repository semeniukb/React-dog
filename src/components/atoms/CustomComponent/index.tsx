export const CustomComponent = ({ variant }: any) => (
  <div>Custom Component 123 {variant}</div>
);
