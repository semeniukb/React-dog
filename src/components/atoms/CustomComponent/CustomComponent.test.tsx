import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import * as stories from './CustomComponent.stories';

describe('CustomComponent', () => {
  const { Primary, Secondary } = composeStories(stories);
  it('should match snapshot Primary', () => {
    const { container } = render(<Primary />);
    expect(container).toMatchSnapshot();
  });
  it('should match snapshot Secondary', () => {
    const { container } = render(<Secondary />);
    expect(container).toMatchSnapshot();
  });
});
