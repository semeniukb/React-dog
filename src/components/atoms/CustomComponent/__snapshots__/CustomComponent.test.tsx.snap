// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomComponent should match snapshot Primary 1`] = `
<div>
  <div>
    Custom Component 123 
    contained
  </div>
</div>
`;

exports[`CustomComponent should match snapshot Secondary 1`] = `
<div>
  <div>
    Custom Component 123 
    outlined
  </div>
</div>
`;
